mlflow/store/dbfs_artifact_repo.py

```diff
@@ -53,17 +22,43 @@ class DbfsArtifactRepository(ArtifactRepository):
     together with the DatabricksStore.
     """
 
-    def __init__(self, artifact_uri, http_request_kwargs):
-        """
-        :param http_request_kwargs arguments to add to rest_utils.http_request for all requests
-            'hostname', 'headers', and 'verify' are required.
-            Should include authentication information to Databricks.
-        """
+    def __init__(self, artifact_uri, get_host_creds):
         cleaned_artifact_uri = artifact_uri.rstrip('/')
         super(DbfsArtifactRepository, self).__init__(cleaned_artifact_uri)
+        self.get_host_creds = get_host_creds
         if not cleaned_artifact_uri.startswith('dbfs:/'):
             raise MlflowException('DbfsArtifactRepository URI must start with dbfs:/')
-        self.http_request_kwargs = http_request_kwargs
+
+    def _databricks_api_request(self, **kwargs):
```

Comment:
Refactored this to move `_databricks_api_request` inside so we can rely on `get_host_creds`. This is similar to the refactoring @smurching did to `projects/databricks.py` -- and that refactoring indeed made this change a lot simpler there.